File: doc/src/scripts/SurfaceSpeed.cpp
#include <iostream>
#include <krpc.hpp>
#include <krpc/services/space_center.hpp>

int main() {
  krpc::Client conn = krpc::connect("Surface speed");
  krpc::services::SpaceCenter spaceCenter(&conn);
  auto vessel = spaceCenter.active_vessel();
  auto ref_frame = vessel.orbit().body().reference_frame();

  while (true) {
    auto velocity = vessel.flight(ref_frame).velocity();
    std::cout
      << "Surface velocity = ("
      << std::get<0>(velocity) << ","
      << std::get<1>(velocity) << ","
      << std::get<2>(velocity)
      << ")" << std::endl;

    auto speed = vessel.flight(ref_frame).speed();
    std::cout << "Surface speed = " << speed << " m/s" << std::endl;

    std::this_thread::sleep_for(std::chrono::seconds(1));
  }
}

File: doc/src/scripts/SurfaceSpeed.cs
using System;
using KRPC.Client;
using KRPC.Client.Services.SpaceCenter;

class SurfaceSpeed
{
    public static void Main ()
    {
        var connection = new Connection (name : "Surface speed");
        var vessel = connection.SpaceCenter ().ActiveVessel;
        var refFrame = vessel.Orbit.Body.ReferenceFrame;

        while (true) {
            var velocity = vessel.Flight (refFrame).Velocity;
            Console.WriteLine ("Surface velocity = ({0:F1}, {1:F1}, {2:F1})",
                               velocity.Item1, velocity.Item2, velocity.Item3);

            var speed = vessel.Flight (refFrame).Speed;
            Console.WriteLine ("Surface speed = {0:F1} m/s", speed);

            System.Threading.Thread.Sleep (1000);
        }
    }
}
